### ISSUE TITLE
Deepin Segmentation Fault

### ISSUE THREAD

**Issue by nodefive** [linux-on-mac/mbpfan#206]
Compiling finishes ok, tests are ok. But when running I get segfault.

```
unix@deepin:~$ uname -r
4.15.0-30deepin-generic
```

```
unix@deepin:~/Desktop/mbpfan$ lsmod | grep -e applesmc -e coretemp
applesmc               20480  0
input_polldev          16384  1 applesmc
coretemp               16384  0
```

```
unix@deepin:~/Desktop/mbpfan$ make
mkdir -p bin
Compiling src/minunit...
cc -c  -g  -Wall -Wextra -Wno-unused-function -std=c99 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=500 src/minunit.c -osrc/minunit.o
mkdir -p bin
Compiling src/settings...
cc -c  -g  -Wall -Wextra -Wno-unused-function -std=c99 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=500 src/settings.c -osrc/settings.o
mkdir -p bin
Compiling src/daemon...
cc -c  -g  -Wall -Wextra -Wno-unused-function -std=c99 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=500 src/daemon.c -osrc/daemon.o
mkdir -p bin
Compiling src/mbpfan...
cc -c  -g  -Wall -Wextra -Wno-unused-function -std=c99 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=500 src/mbpfan.c -osrc/mbpfan.o
mkdir -p bin
Compiling src/main...
cc -c  -g  -Wall -Wextra -Wno-unused-function -std=c99 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=500 src/main.c -osrc/main.o
mkdir -p bin
Compiling src/util...
cc -c  -g  -Wall -Wextra -Wno-unused-function -std=c99 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=500 src/util.c -osrc/util.o
mkdir -p bin
Compiling src/strmap...
cc -c  -g  -Wall -Wextra -Wno-unused-function -std=c99 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=500 src/strmap.c -osrc/strmap.o
Linking...
cc  -g src/minunit.o src/settings.o src/daemon.o src/mbpfan.o src/main.o src/util.o src/strmap.o -lm -o bin/mbpfan
```

```
unix@deepin:~/Desktop/mbpfan$ sudo cp mbpfan.conf /etc/
unix@deepin:~/Desktop/mbpfan$ sudo ./bin/mbpfan -t
Starting the tests..
It is normal for them to take a bit to finish.
Testing the _supplied_ mbpfan.conf (not the one you are using)..
ALL TESTS PASSED
Tests run: 7
```

```
unix@deepin:~/Desktop/mbpfan$ sudo bin/mbpfan 
Segmentation fault
```

**Comment by gaul** [comment]
Can you run mbpfan under valgrind or gdb and get the backtrace?

**Comment by nodefive** [comment]
Sure. Here:

```
Program received signal SIGSEGV, Segmentation fault.
strlen () at ../sysdeps/x86_64/strlen.S:106
106	../sysdeps/x86_64/strlen.S: No such file or directory.
```

**Comment by gaul** [comment]
Thanks for reporting this!  Could you check if the latest master fixes your symptoms?  If so I will release 2.2.1.

**Comment by nodefive** [comment]
Now works just fine :-) Thanks Andrew

```
Starting program: /home/unix/Desktop/mbpfan/bin/mbpfan 
mbpfan 2.2.0 starting up
[Inferior 1 (process 9005) exited normally]
```

```
 unix [~/Desktop/mbpfan] $ psg mbp
 8436 ?        00:00:00 mbpfan
```
